gammapy/datasets/simulate.py

```diff
@@ -85,7 +182,12 @@ def sample_sources(self, dataset):
             else:
                 temporal_model = evaluator.model.temporal_model
 
-            table = self._sample_coord_time(npred, temporal_model, dataset.gti)
+            if (
+                temporal_model == "TemplateTemporalModel"
+            ):  # check the condition when the format model is defined
+                table = self._sample_coord_time_energy(dataset, evaluator)
+            else:
+                table = self._sample_coord_time(npred, temporal_model, dataset.gti)
```

Comment:
Maybe the line computing the `flux` and the `npred` should be moved in this `else` clause to avoid computing `npred` for a time varying spectral model.

Reply:
Thank you @registerrier . I've implemented your suggestion.